src/win/process.c

```diff
@@ -929,7 +937,21 @@ int uv_spawn(uv_loop_t* loop,
   }
 
   if (options->file == NULL ||
-      options->args == NULL) {
+      options->args == NULL ||
+      /*
+       * UV_PROCESS_DETACHED and UV_PROCESS_WINDOWS_RUNAS_ADMIN are
+       * mutually exclusive. If both are set, we return UV_EINVAL.
+       *
+       * stdio, cwd, env and stdio_count are not supported when
+       * UV_PROCESS_WINDOWS_RUNAS_ADMIN is set. If any of these are
+       * set, we return UV_EINVAL.
+       */
+      (((options->flags & UV_PROCESS_DETACHED) ||
+        options->stdio != NULL ||
+        options->cwd != NULL ||
+        options->env != NULL ||
+        options->stdio_count > 0) &&
+       (options->flags & UV_PROCESS_WINDOWS_RUNAS_ADMIN))) {
     return UV_EINVAL;
   }
```

Comment:
The logic here becomes hard to follow. Suggestion: break it up in several `if` statements. I can't post it as a diff but something like:
```c
if (options->file == NULL || options->args == NULL)
  return UV_EINVAL;

if (options->flags & UV_PROCESS_WINDOWS_RUNAS_ADMIN) {
  if (options->flags & UV_PROCESS_DETACHED)
    return UV_EINVAL;
  if (options->stdio != NULL || options->cwd != NULL || options->env != NULL)
    return UV_EINVAL;
  if (options->stdio_count > 0)
    return UV_EINVAL;
}
```

Reply:
got it